generator/generator_tests/relation_tags_tests.cpp

```diff
@@ -0,0 +1,263 @@
+#include "testing/testing.hpp"
+
+#include "generator/generator_tests/common.hpp"
+#include "generator/relation_tags.hpp"
+#include "generator/intermediate_data.hpp"
+
+namespace relation_tags_tests
+{
+using namespace generator_tests;
+using namespace generator;
+using namespace feature;
+using namespace std;
+
+// In memory relations storage (copy-n-paste from collector_building_parts_tests.cpp).
+class TestOSMElementCacheReader : public generator::cache::OSMElementCacheReaderInterface
+{
+public:
+  TestOSMElementCacheReader(std::unordered_map<generator::cache::Key, RelationElement> & m)
+    : m_mapping(m)
+  {
+  }
+
+  // OSMElementCacheReaderInterface overrides:
+  bool Read(generator::cache::Key /* id */, WayElement & /* value */) override { UNREACHABLE(); }
+
+  bool Read(generator::cache::Key id, RelationElement & value) override
```

Comment:
nit: it can be refactored to return std::optional instead of bool and element-by-reference.

Reply:
Switch to `optional` wold be great. But it requires a lot of changes in `intermediate_data.hpp`, `intermediate_data.cpp`, etc. Not sure I am confident enough.